docs/src/components/Hero/SelectedLabel/index.tsx

```diff
@@ -1,38 +1,191 @@
+      height: positionStyles.height + sizeChange.y,
+    });
+
+    // these magic numbers are a result of disparity between font's apparent and actual size
+    const sizeOffsetX = 0.98;
+    const sizeOffsetY = 1.25;
+
+    // scale starts at 1 and as it gets larger approaches sizeOffset
+    // prettier-ignore
+    textScale.x =
+      (positionStyles.width / constantStyles.initialWidth) 
+      * sizeOffsetX - sizeOffsetX + 1;
+    // prettier-ignore
+    textScale.y =
+      (positionStyles.height / constantStyles.initialHeight) 
+      * sizeOffsetY - sizeOffsetY + 1;
+
+    if (textScale.x < 0) textScale.x = 0;
+    if (textScale.y < 0) textScale.y = 0;
+
+    applyTextScale();
+  };
 
-const SelectedLabel: React.FC<{ children: React.ReactNode }> = ({
-  children,
-}) => {
   return (
-    <span className={clsx(styles.headingLabel, styles.selection)}>
-      <div className={styles.selectionContainer}>
-        <div
-          className={clsx(
-            styles.selectionBox,
-            styles.boxUpper,
-            styles.boxLeft
-          )}></div>
-        <div
-          className={clsx(
-            styles.selectionBox,
-            styles.boxUpper,
-            styles.boxRight
-          )}></div>
-        <div
-          className={clsx(
-            styles.selectionBox,
-            styles.boxLower,
-            styles.boxLeft
-          )}></div>
-        <div
-          className={clsx(
-            styles.selectionBox,
-            styles.boxLower,
-            styles.boxRight
-          )}></div>
-        {children}
+    <span
+      ref={selectionRef}
+      className={clsx(styles.headingLabel, styles.selection)}
+      style={{
+        position: constantStyles.isTextInteractive ? 'absolute' : 'relative',
+      }}>
+      <div ref={selectionContainerRef} className={styles.selectionContainer}>
+        <SelectionBox
+          propagationFunction={positionPropagator}
+          draggableIdentifier={DraggableId.TOP_LEFT}
+          isInteractive={isInteractive}></SelectionBox>
+        <SelectionBox
+          propagationFunction={positionPropagator}
+          draggableIdentifier={DraggableId.TOP_RIGHT}
+          isInteractive={isInteractive}></SelectionBox>
+        <SelectionBox
+          propagationFunction={positionPropagator}
+          draggableIdentifier={DraggableId.BOTTOM_LEFT}
+          isInteractive={isInteractive}></SelectionBox>
+        <SelectionBox
+          propagationFunction={positionPropagator}
+          draggableIdentifier={DraggableId.BOTTOM_RIGHT}
+          isInteractive={isInteractive}></SelectionBox>
+        <SelectionBox
+          propagationFunction={positionPropagator}
+          draggableIdentifier={DraggableId.CENTER}
+          isInteractive={isInteractive}>
+          <span
+            ref={textLabelRef}
+            className={clsx(
+              isInteractive ? styles.preInteractiveHeaderText : null,
+              constantStyles.isTextInteractive
```

Comment:
can we depend only on one of those flags? isInteractive os constantStyles.isTextInteractive